src/components/ResponsiveGrid/ResponsiveGrid.tsx

```diff
@@ -0,0 +1,187 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'react-peek/prop-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import {
+	StandardProps,
+	findTypes,
+	omitProps,
+} from '../../util/component-types';
+import Resizer from '../Resizer/Resizer';
+
+const cx = lucidClassNames.bind('&-ResponsiveGrid');
+
+const { string, number, arrayOf } = PropTypes;
+
+interface IResponsiveGridCellProps extends StandardProps {}
+
+const Cell = (_props: IResponsiveGridCellProps): null => null;
+Cell.displayName = 'ResponsiveGrid.Cell';
+Cell.peek = {
+	description: `
+		Renders a \`<span>\` as the grid cell
+	`,
+};
+
+interface IResponsiveGridWrapperProps
+	extends StandardProps,
+		React.DetailedHTMLProps<
+			React.HTMLAttributes<HTMLDivElement>,
+			HTMLDivElement
+		> {
+	/**
+	 * Break points for the grid to switch column layouts. Break points should be provided
+	 * in ascending order. Currently only two break points are used. Example: [960, 1430]
+	 */
+	breakPoints: number[];
+}
+
+interface IResponsiveGridProps extends IResponsiveGridWrapperProps {
+	/**
+	 * Width of the grid. Note: this does not set the width of the grid, and should be
+	 * provided by calculating the width of the parent element.
+	 */
+	width: number;
+}
+
+const defaultProps = {
+	breakPoints: [960, 1430],
+};
+
+export class ResponsiveGrid extends React.Component<IResponsiveGridProps> {
+	static displayName = 'ResponsiveGrid';
+
+	static propTypes = {
+		width: number`Width of the grid. Note: this does not set the width of the grid, and should be
+		provided by calculating the width of the parent element.`,
+
+		breakPoints: arrayOf(number)`
+		Break points for the grid to switch column layouts. Break points should be provided 
+		in ascending order. Currently only two break points are used. Example: [960, 1430]`,
+
+		className: string`
+			Appended to the component-specific class names set on the root elements.
+		`,
+	};
+
+	getColumnLayout = (numberOfColumns: number) => {
+		const cellProps = _.map(
+			findTypes(this.props, ResponsiveGridWrapper.Cell),
+			'props'
+		);
+
+		const columns = _.reduce(
+			/* We use ts-ignore below since the typescript compiler pattern matches this _.map
+			 to an array of booleans */
+			_.map(cellProps, (props: IResponsiveGridCellProps, key: number) => (
+				<article key={key} className={cx('&-Cell')}>
+					{props.children}
+				</article>
+			)),
+			(columns: React.ReactElement[][], cell, idx: number) => {
+				//@ts-ignore
```

Comment:
This issue (and the above comment regarding `_.map`'s return type being `boolean[]`) can be resolved by declaring cellProps with its Interface type on line 68, e.g. `const cellProps: IResponsiveGridCellProps[]`